examples/proxy.rs

```diff
@@ -0,0 +1,47 @@
+#![deny(warnings)]
+extern crate http;
+extern crate hyper;
+extern crate pretty_env_logger;
+
+use http::HttpTryFrom;
+use hyper::{Client, Server};
+use hyper::service::service_fn;
+use hyper::rt::{self, Future};
+use std::net::SocketAddr;
+use std::sync::Arc;
+
+fn main() {
+    pretty_env_logger::init();
+
+    let in_addr = ([127, 0, 0, 1], 3001).into();
+    let out_addr: SocketAddr = ([127, 0, 0, 1], 3000).into();
+
+    let client_arc = Arc::new(Client::new());
+
+    let out_addr_clone = out_addr.clone();
+    // new_service is run for each connection, creating a 'service'
+    // to handle requests for that specific connection.
+    let new_service = move || {
+        let client = client_arc.clone();
+        // This is the `Service` that will handle the connection.
+        // `service_fn_ok` is a helper to convert a function that
+        // returns a Response into a `Service`.
+        service_fn(move |mut req| {
+            let uri_string = format!("http://{}/{}",
+                out_addr_clone,
+                req.uri().path_and_query().map(|x| x.as_str()).unwrap_or(""));
+            let uri = HttpTryFrom::try_from(uri_string).unwrap();
```

Comment:
Instead of the odd trait (`HttpTryFrom` will go away someday, whenever `TryFrom` actually stabilizes (so like 2038?)), this change just be `uri_string.parse()`.

Reply:
I'll give that a try.